File: 2/2000s/28s/2843/CountSymmetricIntegers.java
/*
https://leetcode.com/problems/count-symmetric-integers/description/?envType=daily-question&envId=2025-04-11
*/

class Solution {

    public static boolean isASymmetricInteger(int val){
        String toCheckString=String.valueOf(val);
        int lengthOfStr=toCheckString.length();
        if(lengthOfStr%2!=0) return false;
        int halfLength=lengthOfStr/2, sumLeft=0, sumRight=0, i=0, j=halfLength;
        while(true){
            if(i>=halfLength||j>=lengthOfStr) break;
            sumLeft+=(toCheckString.charAt(i++)-'0');
            sumRight+=(toCheckString.charAt(j++)-'0');
        }
        return sumLeft==sumRight;
    }

    public int countSymmetricIntegers(int low, int high) {
        int resultantSymmNums=0;
        while(low<=high){
            if(isASymmetricInteger(low)){
                resultantSymmNums++;
            }
            if(low==high) break;
            if(isASymmetricInteger(high)){
                resultantSymmNums++;
            }
            low++;
            high--;
        }
        return resultantSymmNums;
    }
}

File: 2/2000s/28s/2843/CountSymmetricIntegers.py
# https://leetcode.com/problems/count-symmetric-integers/description/?envType=daily-question&envId=2025-04-11

class Solution:
    def countSymmetricIntegers(self, low: int, high: int) -> int:

        def isASymmetricInteger(value):
            toCheckString=str(value)
            lengthOfStr=len(toCheckString)
            if lengthOfStr%2!=0:
                return False
            halfLength=lengthOfStr//2
            sumLeft=0
            sumRight=0
            i=0
            j=halfLength
            while(True):
                if i>=halfLength and j>=lengthOfStr:
                    break
                sumLeft+=int(toCheckString[i])
                sumRight+=int(toCheckString[j])
                i+=1
                j+=1
            if sumLeft==sumRight:
                return True
            return False

        resultantSymmNums=0
        while(low<=high):
            if isASymmetricInteger(low):
                resultantSymmNums+=1
            if low==high:
                break;
            if isASymmetricInteger(high):
                resultantSymmNums+=1
            low+=1
            high-=1
        return resultantSymmNums